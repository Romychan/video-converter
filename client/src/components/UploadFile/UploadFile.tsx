import { Button } from '@ui/Button/Button';
import { Dropzone } from '@ui/Dropzone/Dropzone';
import { Icon } from '@ui/Icon/Icon';

interface IUploadFileProps {
  uploadFile: (file: FileList) => void;
  types: string[];
}

export const UploadFile = ({ uploadFile, types }: IUploadFileProps) => {
  return (
    <Dropzone
      title="Перетащите видеофайл в эту область"
      icon={<Icon name="upload-video" />}
      uploadFileHandler={uploadFile}
      acceptedTypes={types}
    >
      <Button state="primary">Выбрать файл</Button>
    </Dropzone>
  );
};
