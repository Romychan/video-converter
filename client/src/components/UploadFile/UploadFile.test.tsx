import {
  fireEvent,
  screen,
  render,
  waitFor,
  RenderResult,
} from '@testing-library/react';
import { vi } from 'vitest';

import { UploadFile } from './UploadFile';

describe('UploadFile', () => {
  const handleChange = vi.fn();
  let container: RenderResult;

  beforeEach(() => {
    container = render(
      <UploadFile types={['mp4']} uploadFile={handleChange} />,
    );
  });

  it('should render correctly', () => {
    expect(screen.getByText(/Перетащите видео/i)).toBeInTheDocument();
    expect(screen.getByText(/Выбрать файл/i)).toBeInTheDocument();
  });

  it('should upload file', async () => {
    const file = new File(['Filename'], 'Filename.mp4', { type: 'video/mp4' });
    const dropzoneInput = screen.getByTestId(
      'dropzone-input',
    ) as HTMLInputElement;

    await waitFor(() =>
      fireEvent.change(dropzoneInput, {
        target: { files: [file] },
      }),
    );

    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(dropzoneInput.files?.[0]).toStrictEqual(file);
    expect(dropzoneInput.files).toHaveLength(1);
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
