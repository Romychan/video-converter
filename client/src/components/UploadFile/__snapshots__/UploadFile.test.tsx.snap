// Vitest Snapshot v1

exports[`UploadFile > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_c96111"
      >
        <div
          class="_dropzone_c96111"
          data-testid="dropzone-container"
        >
          <input
            accept=".mp4"
            class="_input_c96111"
            data-testid="dropzone-input"
            type="file"
          />
          <div
            class="_icon_c96111"
          >
            <svg
              class="icon icon-upload-video"
              data-testid="icon"
              height="16"
              stroke=""
              width="16"
            >
              <use
                xlink:href="/src/assets/img/general/icons.svg#upload-video"
              />
            </svg>
          </div>
          <p
            class="subtitle _subtitle_c96111"
          >
            Перетащите видеофайл в эту область
          </p>
          <p
            class="text text_small _text_c96111"
          >
            или нажмите на кнопку
          </p>
          <button
            class="_button_b22f72 _button_c96111 _primary_b22f72"
          >
            Выбрать файл
          </button>
        </div>
        <div
          class="_background_c96111"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_c96111"
    >
      <div
        class="_dropzone_c96111"
        data-testid="dropzone-container"
      >
        <input
          accept=".mp4"
          class="_input_c96111"
          data-testid="dropzone-input"
          type="file"
        />
        <div
          class="_icon_c96111"
        >
          <svg
            class="icon icon-upload-video"
            data-testid="icon"
            height="16"
            stroke=""
            width="16"
          >
            <use
              xlink:href="/src/assets/img/general/icons.svg#upload-video"
            />
          </svg>
        </div>
        <p
          class="subtitle _subtitle_c96111"
        >
          Перетащите видеофайл в эту область
        </p>
        <p
          class="text text_small _text_c96111"
        >
          или нажмите на кнопку
        </p>
        <button
          class="_button_b22f72 _button_c96111 _primary_b22f72"
        >
          Выбрать файл
        </button>
      </div>
      <div
        class="_background_c96111"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
